docs/source/usage/workflow.rst

```diff
@@ -0,0 +1,30 @@
+.. _workflow:
+
+Usual workflow
+==============
+
+Deferred operation mode
+-----------------------
+
+IO operations are in general not performed by the openPMD API immediately after calling the corresponding API function.
+Rather, operations are enqueued internally and performed at so-called *flush points*.
+A flush point is a point within an application's sequential control flow where the openPMD API must uphold the following guarantees:
+
+*   In write mode, any change made to a user buffer whose data shall be stored in a dataset up to the flush point must be found in the written dataset.
+*   In write mode, no change made to a user buffer whose data shall be stored in a dataset after the flush point must be found in the written dataset.
+*   In read mode, a buffer into which data from a dataset should be filled, must not be altered by the openPMD API before the flush point.
+*   In read mode, a buffer into which data from a dataset should be filled, must have been filled with the requested data after the flush point.
+
+In short: ``storeChunk()`` and ``loadChunk()`` operations must happen exactly at flush points.
+
+Flush points are triggered by:
+
+*   Calling ``Series::flush()``.
+*   Calling ``Iteration::flush( flush=true )``.
+    Flush point guarantees affect only the corresponding iteration.
+*   The streaming API (i.e. ``Series.readIterations()`` and ``Series.writeIteration()``) automatically before accessing the next iteration.
+
+Attributes are (currently) unaffected by this:
+
+*   In writing, attributes are stored by value and can afterwards not be aliased by the user.
```

Comment:
What is meant with aliased? :sweat_smile:

Reply:
(Does anything change for ADIOS2 where attributes become vars in the new layout? Probably not, just double-checking.)

Reply:
> What is meant with aliased? sweat_smile

After `defineAttribute`, the attribute values are store some place internal that a user has no access to, so a user cannot have a pointer to the value or something. "aliased" in the same meaning as C `noalias`. The openPMD API can deal with attributes whatever way it wishes to do, since it's entirely transparent to the user.

> (Does anything change for ADIOS2 where attributes become vars in the new layout? Probably not, just double-checking.)

No, since these arguments are entirely about frontend things. Attributes to be written in the new layout are buffered in `struct BufferedAttributeWrite`.